### DISCUSSION TITLE
custom select component not working with controller

### DISCUSSION THREAD

**Discussion by niralparmar** [react-hook-form/react-hook-form#1820] [Q&A] [answered]
hi,
i have a select component which is a wrapper of react-select and i passed it to controller but i dont see any data getting submitted
my codesanbox
[https://codesandbox.io/s/musing-glitter-0brjd](url)

in general, what would be correct way to implement wrapper component on top of other component library and pass it to controller.

for eg: here is my wrapper component for reactstrap input
`
                  
                 import React from "react"; 
                 import { Input} from 'reactstrap';
                 export default function TextBox({
                       id,
                       name,
                       label,
                       value = "",
                       placeholder = "",
                       readonly = false,
                       hidden = false,
                       required = false,
                       isValid
                   }) {
                  if (hidden) {
                     return null;
                    }
                 return (
                            <Input type="text" name={name} id={id} value={value} placeholder={placeholder}  disabled={readonly} valid={isValid} invalid={!isValid} />
                );
        }
`

**Comment by bluebill1049** [comment, marked as answer]
provide a codesandbox for your issue.

**Reply by niralparmar** [reply]
@bluebill1049  here it is
https://codesandbox.io/s/musing-glitter-0brjd

**Reply by bluebill1049** [reply]
there is an example under: https://react-hook-form.com/api#Controller
https://codesandbox.io/s/react-hook-form-controller-079xx

**Reply by niralparmar** [reply]
thanks @bluebill1049  i followed the exact steps from example , but mine is not working

**Reply by bluebill1049** [reply]
`const MyComponent = () => <Select options={options} />; `

you didn't propagate all the props.

**Reply by niralparmar** [reply]
dang.... thanks @bluebill1049
